handler/ratelimit/limiter.go

```diff
@@ -77,51 +72,29 @@ func (l *Limiter) slowTripper() {
 		case <-l.limits[0].quitCh:
 			return
 		case trip := <-l.check:
-			select {
-			case <-trip.req.Context().Done():
-				// The request was canceled while in the queue.
-				trip.err = trip.req.Context().Err()
-				trip.out <- trip
-
-				// Do not sleep for X canceled requests.
-				continue
-			default:
-			}
-
-			l.mu.Lock()
-
-			if mode, timeToWait := l.checkCapacity(); mode == modeBlock && timeToWait > 0 {
+			mode, timeToWait := l.checkCapacity()
+			if mode == modeBlock && timeToWait > 0 {
 				// We do not wait, we want block directly.
 				trip.err = errors.BetaBackendRateLimitExceeded
 				trip.out <- trip
-
-				l.mu.Unlock()
 			} else {
 				select {
-				// Noop if 'timeToWait' is 0.
 				case <-time.After(timeToWait):
+					// Noop if 'timeToWait' is 0.
 				case <-trip.req.Context().Done():
 					// The request was canceled while in the queue.
 					trip.err = trip.req.Context().Err()
 					trip.out <- trip
 
 					// Do not sleep for X canceled requests.
-					l.mu.Unlock()
 					continue
 				}
 
 				l.countRequest()
 
-				l.mu.Unlock()
-
 				// Do not wait for the response...
 				go func() {
 					trip.res, trip.err = l.transport.RoundTrip(trip.req)
-
-					if trip.res != nil && trip.res.StatusCode == http.StatusTooManyRequests {
-						trip.err = errors.BetaBackendRateLimitExceeded.With(trip.err)
-					}
```

Comment:
this is a behaviour change. An upstream `429` (which wasn't necessarily throttled at all) currently results in a _couper error_. While that still results in a status `429` response to the client, any couper error handler will not trigger anymore. Usually this will mean that the upstream response body is passed through `proxy` endpoints - instead of the default couper error page.

I still think that this current behaviour is unexpected. So I'm fine with the change. but we should add a _change_ type changelog entry.